permissions_validators/src/private_blockchain/mod.rs

```diff
@@ -38,3 +38,541 @@ impl<W: WorldTrait> IsGrantAllowed<W> for ProhibitGrant {
+
+            assert!(only_accounts_data.check(&alice_id, &op, &wsv).is_err());
+        }
+    }
+
+    #[test]
+    fn find_all_assets_definitions() {
+        let TestEnv { alice_id, wsv, .. } = TestEnv::new();
+
+        let op = QueryBox::FindAllAssetsDefinitions(FindAllAssetsDefinitions::new());
+
+        {
+            let only_accounts_domain: IsQueryAllowedBoxed<World> = query::OnlyAccountsDomain.into();
+
+            assert!(only_accounts_domain.check(&alice_id, &op, &wsv).is_err());
+        }
+
+        {
+            let only_accounts_data: IsQueryAllowedBoxed<World> = query::OnlyAccountsData.into();
+
+            assert!(only_accounts_data.check(&alice_id, &op, &wsv).is_err());
+        }
+    }
+
+    #[test]
+    fn find_asset_by_id() {
+        let TestEnv {
+            alice_id,
+            carol_id,
+            wsv,
+            gold_asset_id,
+            silver_asset_id,
+            bronze_asset_id,
+            ..
+        } = TestEnv::new();
+
+        let op1 = QueryBox::FindAssetById(FindAssetById::new(gold_asset_id));
+        let op2 = QueryBox::FindAssetById(FindAssetById::new(silver_asset_id));
+        let op3 = QueryBox::FindAssetById(FindAssetById::new(bronze_asset_id));
+
+        {
+            let only_accounts_domain: IsQueryAllowedBoxed<World> = query::OnlyAccountsDomain.into();
+
+            assert!(only_accounts_domain.check(&alice_id, &op1, &wsv).is_ok());
+            assert!(only_accounts_domain.check(&alice_id, &op2, &wsv).is_ok());
+            assert!(only_accounts_domain.check(&carol_id, &op3, &wsv).is_ok());
+            assert!(only_accounts_domain.check(&alice_id, &op3, &wsv).is_err());
+        }
+
+        {
+            let only_accounts_data: IsQueryAllowedBoxed<World> = query::OnlyAccountsData.into();
+
+            assert!(only_accounts_data.check(&alice_id, &op1, &wsv).is_ok());
+            assert!(only_accounts_data.check(&alice_id, &op2, &wsv).is_err());
+            assert!(only_accounts_data.check(&carol_id, &op3, &wsv).is_ok());
+            assert!(only_accounts_data.check(&alice_id, &op3, &wsv).is_err());
+        }
+    }
+
+    #[test]
+    fn find_asset_definition_by_id() {
+        let TestEnv {
+            alice_id,
+            carol_id,
+            wsv,
+            gold_asset_definition_id,
+            silver_asset_definition_id,
+            bronze_asset_definition_id,
+            ..
+        } = TestEnv::new();
+
+        let op1 = QueryBox::FindAssetDefinitionById(FindAssetDefinitionById::new(
+            gold_asset_definition_id,
+        ));
+        let op2 = QueryBox::FindAssetDefinitionById(FindAssetDefinitionById::new(
+            silver_asset_definition_id,
+        ));
+        let op3 = QueryBox::FindAssetDefinitionById(FindAssetDefinitionById::new(
+            bronze_asset_definition_id,
+        ));
+
+        {
+            let only_accounts_domain: IsQueryAllowedBoxed<World> = query::OnlyAccountsDomain.into();
+
```

Comment:
This is an example of a good assertion block. You need to be able to demonstrate when it works and when it doesn't.